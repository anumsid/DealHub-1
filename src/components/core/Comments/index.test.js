import React from 'react'
import { shallow } from 'enzyme'
import { stub } from 'sinon'
import toJson from 'enzyme-to-json'
import { Comments } from './index'

describe('Comments Component', () => {
  let wrapper
  let onClick
  beforeEach(() => {
    onClick = stub()
    const props = {
      authToken: 'xxx',
      onClick: onClick,
      comments: [{
        content: 'hello world',
        user_id: 1,
        id: 2
      },
      {
        content: 'this is cool!',
        user_id: 1,
        id: 3
      }],
      users: [{
        id: 1,
        name: 'joe blow'
      }]
    }
    wrapper = shallow(<Comments {...props} />)
  })

  it('Matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('Handles click on form submit', () => {
    wrapper.find('form').simulate('submit', {preventDefault: () => {}})
    expect(onClick.calledOnce).toBe(true)
  })
})
