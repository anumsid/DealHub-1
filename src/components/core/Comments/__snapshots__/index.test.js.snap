// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comments Component Matches snapshot 1`] = `
<div
  className="comments"
>
  <form
    onSubmit={[Function]}
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    <input
      name="content"
      onChange={[Function]}
      placeholder="Add Comment"
      type="text"
    />
    <label
      htmlFor="Add Comment"
    />
    <button>
      Add Comment!
    </button>
  </form>
  <div
    className="ui divided middle aligned list"
  >
    <MapComments
      comments={
        Array [
          Object {
            "content": "hello world",
            "id": 2,
            "user_id": 1,
          },
          Object {
            "content": "this is cool!",
            "id": 3,
            "user_id": 1,
          },
        ]
      }
      users={
        Array [
          Object {
            "id": 1,
            "name": "joe blow",
          },
        ]
      }
    />
  </div>
</div>
`;
